packages/webpack-extraction-plugin/src/transformSync.ts

```diff
@@ -0,0 +1,57 @@
+import * as Babel from '@babel/core';
+import { babelPluginStripGriffelRuntime, StripRuntimeBabelPluginMetadata } from './babelPluginStripGriffelRuntime';
+
+export type TransformOptions = {
+  filename: string;
+
+  inputSourceMap: Babel.TransformOptions['inputSourceMap'];
+  enableSourceMaps: boolean;
+};
+
+export type TransformResult = {
+  code: string;
+  cssRules: string[] | undefined;
+  sourceMap: NonNullable<Babel.BabelFileResult['map']> | undefined;
+};
+
+/**
+ * Transforms passed source code with Babel, uses user's config for parsing, but ignores it for transforms.
+ */
+export function transformSync(sourceCode: string, options: TransformOptions): TransformResult {
```

Comment:
This is more or less the same as in existing loader. Once the POC will be done I will refactor it to a separate package.